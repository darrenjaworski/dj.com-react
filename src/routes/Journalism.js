import React from "react";

import journalismData from "../data/journalism.json";
import { Heading, SmallHeading } from "../components/heading";
import { TextLink } from "../components/links";
import { ListItem, UnorderedList } from "../components/list";

const Journalism = () => {
  let sections = journalismData.map(d => {
    let articles = d.articles.map(a => {
      return (
        <ListItem key={a.title.replace(" ", "-")}>
          <TextLink href={a.link}>{a.title}</TextLink>
        </ListItem>
      );
    });
    return (
      <React.Fragment key={d.section.replace(" ", "-")}>
        <SmallHeading>{d.section}</SmallHeading>
        <UnorderedList>{articles}</UnorderedList>
      </React.Fragment>
    );
  });
  return (
    <>
      <Heading>journalism</Heading>
      {sections}
    </>
  );
};

export default Journalism;
