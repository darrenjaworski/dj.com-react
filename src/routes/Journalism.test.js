import React from "react";
import renderer from "react-test-renderer";

import Journalism from "./Journalism";

describe("FourOhFour", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<Journalism />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
