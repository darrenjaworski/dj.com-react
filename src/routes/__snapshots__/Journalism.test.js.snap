// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FourOhFour renders without crashing 1`] = `
Array [
  <h1
    className="css-lozscu"
  >
    journalism
  </h1>,
  <h1
    className="css-wqkrd8"
  >
    oklahoma watch
  </h1>,
  <ul
    className="css-g1maxw"
  >
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/07/22/oklahomas-correctional-facilities/"
      >
        Oklahoma Correctional Facilities
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/07/24/the-mcalester-prison-riot/"
      >
        Big Mac Riot of 73
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/07/22/oklahoma-state-penitentiary/"
      >
        Oklahoma State Penitentiary
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/07/26/charting-inmate-counts/"
      >
        Oklahoma Inmate Counts
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/07/20/oklahoma-watch-study-identifies-jobs-with-most-uninsured-workers/"
      >
        OK Watch Study Identifies Uninsured Workers
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/07/17/the-uninsured/"
      >
        Where the Uninsured Work
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/07/10/permits-for-handgun-licenses-keep-rising/"
      >
        Handgun Permits
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/06/24/cost-of-attendance/"
      >
        Cost of Attendance
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/05/24/where-tornados-strike/"
      >
        Where Tornadoes Strike (article)
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://darrenjaworski.cartodb.com/tables/tornadoes90_13/embed_map?title=false&amp;description=false&amp;search=false&amp;shareable=false&amp;cartodb_logo=true&amp;scrollwheel=false&amp;sql=&amp;sw_lat=33.76088200086917&amp;sw_lon=-103.86474609375&amp;ne_lat=37&amp;ne_lon=-93.31787109374999&amp;height=600&amp;id=cartodb-1369334401177"
      >
        Where Tornadoes Strike (working graphic)
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/05/10/data-driven-spin-the-wheel-to-analyze-the-state-budget/"
      >
         Interactive Sunburst Partition of the State Budget
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/04/02/data-center-profile-of-oklahoma-legislature/"
      >
        Interactive of the State Legislature (article)
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://games.darrenjaworski.com/legislative-graphic/"
      >
        Interactive of the State Legislature (working graphic)
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/02/19/where-the-states-stand-on-medicaid-expansion/"
      >
        Where the States Stand on Medicaid Expansion
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/04/17/mapping-obesity-in-oklahoma/"
      >
        Mapping Obesity in Oklahoma
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/05/08/what-oklahoma-hospitals-charge/"
      >
        Comparing What Oklahoma Hospitals Charge
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/04/26/data-driven-locating-the-uninsured/"
      >
        Locating the Uninsured
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="ttp://oklahomawatch.org/2013/03/27/reading-test-results-for-oklahoma-schools-and-districts/"
      >
        Reading Test Results for Oklahoma Schools and Districts
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/03/15/oklahoma-lake-levels-decline/"
      >
        Where Lake Levels Stand
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/03/01/making-short-work-of-an-average-life/"
      >
        Making Short Work of an Average Life
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/02/20/ou-and-osu-graduation-rates-comparison/"
      >
        The Growing, but Still Elusive, 4-Year College Degree
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2013/01/29/firearm-deaths-in-ok-and-us/"
      >
        Firearm Deaths in Oklahoma
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://oklahomawatch.org/2012/11/09/visualizations-of-ok-hispanic-population/"
      >
        Visualizations of Oklahomas Hispanic Population
      </a>
    </li>
  </ul>,
  <h1
    className="css-wqkrd8"
  >
    routes
  </h1>,
  <ul
    className="css-g1maxw"
  >
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://routes.ou.edu/?p=467"
      >
        Captain Separated from Family for 10 years
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://routes.ou.edu/?p=170"
      >
        Challenging Oklahoma Voter ID Law
      </a>
    </li>
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="http://routes.ou.edu/?p=167"
      >
        Voter identification laws strengthened
      </a>
    </li>
  </ul>,
  <h1
    className="css-wqkrd8"
  >
    journalism master's research
  </h1>,
  <ul
    className="css-g1maxw"
  >
    <li
      className="css-z4ydkm"
    >
      <a
        className="css-alkraz"
        href="https://www.dropbox.com/s/vmj3cxm8wa1l5xm/sciencecommunication.pdf?dl=0"
      >
        Online Science Communication and Public Science Literacy
      </a>
    </li>
  </ul>,
]
`;
